src/components/AddressSearch/index.js

```diff
@@ -254,6 +254,17 @@ function AddressSearch(props) {
             values.state = stateFallback;
         }
 
+        // Some edge-case addresses may lack both street_number and route in the API response, resulting in an empty "values.street"
+        // We are setting up a fallback to ensure "values.street" is populated with a relevant value
+        if (!values.street && details.adr_address) {
+            const streetAddressRegex = /<span\s+class\s*=\s*"street-address"[^>]*>([^<]*)<\/span>/;
```

Comment:
```suggestion
            const streetAddressRegex = /<span class="street-address">([^<]*)<\/span>/;
```
I've not tested this yet, but I wonder how necessary all these extra whitespace checks are. Since we've agreed this will follow a standardised format, I think we can do a straight check for the exact span with this specific class.

Reply:
I just thought it wouldn't hurt to be extra careful but yeah sure since it's a standardised format then the white space checks are not a must or anything.